lib/ui/pages/news_list/primary_news_list_item.dart

```diff
@@ -0,0 +1,116 @@
+import 'package:allthenews/ui/pages/news_list/primary_news_list_entity.dart';
+import 'package:flutter/material.dart';
+
+abstract class _Constants {
+  static const double newsImageRadius = 15;
+  static const double newsImageLayerOpacity = 0.7;
+  static const double authorImageRadius = 15;
+  static const double authorRightPadding = 8;
+  static const double stackItemPadding = 20;
+}
+
+class PrimaryNewsListItem extends StatelessWidget {
+  final PrimaryNewsListEntity news;
+
+  PrimaryNewsListItem({Key key, @required this.news})
+      : assert(news != null),
+        super(key: key);
+
+  @override
+  Widget build(BuildContext context) => Stack(
```

Comment:
generalnie we flutterze parent poprzez ograniczenia (lub ich brak) wielkości może wpłynąć na child widgeta przez co w jakiś tam sposób trzeba o tym pamiętać projektując widget - tutaj jeżeli parent pozwoliłby żeby stack zajął całą dostępną powierzchnię to listItem się rozjeżdża (ten mój tak samo się rozjeżdżał, więc musiałem go nieco poprawić :P) - przykładowo jak owrapujemy ten stack w expanded to tekst rozjedzie się ze zdjęciem - nie wiem jakie jest właściwe podejście przy projektowaniu widgetów, wydaje mi się, że fajnie jakby dało radę zrobić to w ten sposób, aby PrimaryNewsListItem był odporny na tego typu side effecty

Reply:
hmm, na tę chwilę ciężko mi podejść do tego, problem zapewne pojawi się dopiero przy wrzucaniu tych itemów do listy i stąd te FIXME wszędzie, bo jestem przygotowany, że będzie trzeba tego typu rzeczy poprawić. Możemy póki co zostawić to tak jak jest i zajmę się skalowaniem tego widgetu przy wrzucaniu do listy?

Reply:
co do projektowania widgetów to myślę, że potrzeba po prostu doświadczenia w tym i może zapisywania jakichś reguł, które zawsze występują. Myślałem o jakimś wspólnym docu gdzie byśmy zbierali wszystkie rzeczy, o których warto pamiętać, może też jakieś gotowce rozwiązań powtarzających się zagadnień, co Ty na to?

Reply:
Pasuje ;)